comprobo_final_project/scripts/chromosome.py

```diff
@@ -82,17 +83,20 @@ def mutate(self, mutation_rate_multiplier=1.0):
                 gene *= random.uniform(0.8, 1.2)
             mutated_genes.append(round(gene, 3))
 
-        # Replace the chromosome's genes with the mutated genes
-        self.genes = mutated_genes
+        # Create new chromosome with genes from the mutated genes
+        return Chromosome(mutated_genes, self.natural_selection)
 
 
     def get_fitness(self):
         """
         Calculate the fitness of a specified chromosome.
         """
 
-        # TODO: implement get_fitness function
-        pass
+        self.natural_selection.use_genes(self.genes)
+        xpos, ypos = self.natural_selection.run()
+        fitness = np.sqrt(xpos**2 + ypos**2)
+        self.natural_selection.reset()
+        return fitness
```

Comment:
This method could use more comments. Especially around what `natural_selection` is doing. If I can't read the actual code like English, then comment it!